packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt

```diff
@@ -296,6 +299,140 @@ class GraphQLApiUnitTests {
     }
 
+    @Test
+    fun test_subscribe_success_event() {
+        flutterPlugin = AmplifyApiPlugin(eventHandler = mockStreamHandler)
+        val testRequest = HashMap<String, Any>()
+        val id = "someCode"
+
+        testRequest["document"] = ("mutation MyMutation(\$name: String!) {"
+                + "createBlog(input: {name: \$name}) {"
+                + "id"
+                + "name"
+                + "createdAt"
+                + "}"
+                + "}")
+
+        testRequest["variables"] = mapOf(
+                "name" to "Test App Blog"
+        )
+        testRequest["cancelToken"] = id
+
+
+
+        val graphQLResponse: GraphQLResponse<String> = GraphQLResponse("result data", null)
+
+        doAnswer { invocation ->
+            Assert.assertEquals(
+                    SimpleGraphQLRequest<String>(
+                            testRequest["document"] as String,
+                            testRequest["variables"] as Map<String, Any>,
+                            String::class.java,
+                            GsonVariablesSerializer()
+                    ),
+                    invocation.arguments[0]
+            )
+            (invocation.arguments[2] as Consumer<GraphQLResponse<String>>).accept(
+                    graphQLResponse
+            )
+            mockGraphQLOperation
+        }.`when`(mockApi).subscribe(
+                any<GraphQLRequest<String>>(),
+                any<Consumer<String>>(),
+                any<Consumer<GraphQLResponse<String>>>(),
+                any<Consumer<ApiException>>(),
+                any<Action>()
+        )
+
+        flutterPlugin.onMethodCall(
+                MethodCall("subscribe", testRequest),
+                mockResult
+        )
+
+        verify(mockStreamHandler, times(1))
+                .sendEvent(
+                        graphQLResponse.data,
+                        graphQLResponse.errors,
+                        id,
+                        GraphQLSubscriptionEventTypes.DATA)
+    }
+
+    @Test
+    fun test_subscribe_malformed_request_error() {
+        val testRequest = HashMap<String, Any>()
+
+        flutterPlugin.onMethodCall(
+                MethodCall("subscribe", testRequest),
+                mockResult
+        )
+
+        verify(mockResult).error(
+                "ApiException",
+                ExceptionMessages.defaultFallbackExceptionMessage,
+                mapOf(
+                        "message" to ExceptionMessages.missingExceptionMessage,
+                        "recoverySuggestion" to ExceptionMessages.missingRecoverySuggestion,
+                        "underlyingException" to "AmplifyException{message=The graphQL document request argument was not passed as a String, cause=kotlin.TypeCastException: null cannot be cast to non-null type kotlin.String, recoverySuggestion=The request should include the graphQL document as a String}"
+                )
+        )
+    }
+
+    @Test
+    fun test_subscribe_api_error() {
```

Comment:
I believe `sendError` scenario is not tested, i.e. when subscription is established and then error is received (An error scenario more likely to happen)

Same for iOS.

Reply:
Done for both iOS and Android.